### ISSUE TITLE
False positive on Naming/MemoizedInstanceVariableName

### ISSUE THREAD

**Issue by halostatue** [rubocop-hq/rubocop#6710]
Received Naming/MemoizedInstanceVariableName incorrectly for methods and variables with surrounding underscores.

--------

## Expected behavior

No Naming/MemoizedInstanceVariableName warning.

## Actual behavior

Received message

> Naming/MemoizedInstanceVariableName: Memoized variable `@__instances__` does not match method name `__instances__`. Use `@__instances__` instead.

## Steps to reproduce the problem

Run RuboCop over:

```ruby
require 'set'

class Foo
  def __instances__
    @__instances__ ||= Set.new
  end
end
```

## RuboCop version

0.63.1 (using Parser 2.5.1.0, running on ruby 2.6.0 x86_64-darwin18)

**Comment by pocke** [comment]
Thank you for your reporting!



More small code to reproduce:

```ruby
def _a
  @_a ||= x
end
```

This cop does not work well if the method name starts with underscore.



related: https://github.com/rubocop-hq/rubocop/pull/6125
